crates/matrix-sdk-crypto/src/utilities.rs

```diff
@@ -21,3 +29,42 @@ where
     let json = serde_json::to_string(value)?;
     serde_json::from_str(&json)
 }
+
+const ISO8601_WITH_MILLIS: iso8601::EncodedConfig = iso8601::Config::DEFAULT
+    .set_time_precision(iso8601::TimePrecision::Second { decimal_digits: NonZeroU8::new(3) })
+    .encode();
+
+/// Format the given timestamp into a human-readable timestamp.
+///
+/// The result looks like `1970-01-01T00:00:00.000Z`.
+pub fn timestamp_to_iso8601(ts: MilliSecondsSinceUnixEpoch) -> String {
+    let nanos_since_epoch = i128::from(ts.get()) * 1_000_000;
+
+    // OffsetDateTime has a max year of 9999, whereas MilliSecondsSinceUnixEpoch has
+    // a max year of 285427, so this can overflow for very large timestamps. (The
+    // Y10K problem!)
+    match OffsetDateTime::from_unix_timestamp_nanos(nanos_since_epoch) {
+        Err(_) => "<out of range>".to_owned(),
```

Comment:
It's a bad API smell that the utility function decides what to do in case of errors. Instead, could we have this method return `Option<String>`, and in this branch it could return `None`?

(or more concisely:

```rust
OffsetDateTime::from_unix_timestamp_nanos(nanos_since_epoch).ok()?.format(&...)
```

Reply:
I think the duplication would be OK, I also get the function is an internal utility. But having it return `Option` could also make it so that we avoid the unwrap below, so I'm tempted to insist just a bit on returning an `Option` here, even though overall I don't have a very strong opinion.